finat/cube.py

```diff
@@ -49,6 +49,13 @@ def _entity_support_dofs(self):
     def entity_dofs(self):
         return self._entity_dofs
 
+    @cached_property
+    def _entity_permutations(self):
+        return flatten_permutations(self.product.entity_permutations())
+
+    def entity_permutations(self):
+        return self._entity_permutations
```

Comment:
Do this once and use `functools.cache` if you think it needs caching.

Reply:
So I would just `@cached_property` this.

Reply:
Fixed.